[youdonnnn] iP

youdonnnn opened the pull request:
### Introducing the YoudonBot!
 > Sign up for a free trial now! [link here](https://en.wikipedia.org/wiki/Cat)

YoudonBot allows you to focus on the more important things in life. We are: 
* text-based
* easy to learn
* ~~FAST~~ _SUPER_ FAST to use

All you need to do ___right now___ is:
1. download it [here](https://en.wikipedia.org/wiki/Dog).
2. double-click the __.exe__ file.
3. add in your tasks and events.
4. and let it do the rest of the _magic_ for you &#x1f604;

And the best part is that it is __FREE__!

Features include:
- [x] Managing tasks
- [x] Managing deadlines (new release!)
- [ ] Reminders (coming soon)
***
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Level 0. Rename, Greet, Exit

Commit:
Level 1. Echo

Commit:
Level 2. Add, List

Commit:
Level 3. Mark as Done

Commit:
Level 4. ToDos, Events, Deadlines

Commit:
Automated Text UI Testing

Commit:
Level 5. Handle Errors

Commit:
Level 6. Delete

Commit:
Use Enums

jieqiboh commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,17 @@
+public class Deadline extends Task {
```
Perhaps a Javadoc comment describing the class would be useful 😃 
I noticed the same issue in other places too

jieqiboh commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,17 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
I like the way you named this boolean variable 😃

jieqiboh commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,19 @@
+
+    @Override
+    public String toString() {
+        return "[" + this.getTypeIcon() + "][" + this.getStatusIcon() + "] " + this.description + "(from: " + this.start + "to:" + this.end + ")";
```
```suggestion
        return "[" + this.getTypeIcon() + "][" + this.getStatusIcon() + "] " 
        + this.description + "(from: " + this.start + "to:" + this.end + ")";
```
I think splitting the code into a newline would be useful 😃

jieqiboh commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,19 @@
+    protected String start;
+    protected String end;
+
+    public Event(String description,  String start, String end) {
```
```suggestion
    public Event(String description, String start, String end) {
```
I think there was a minor typo where you used 2 spaces to separate non-whitespace characters instead of 1

jieqiboh commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,17 @@
+
+    @Override
+    public String toString() {
+        return "[" + this.getTypeIcon() + "][" + this.getStatusIcon() + "] " + this.description + "(by: " + this.deadline + ")";
```
```suggestion
        return "[" + this.getTypeIcon() + "][" + this.getStatusIcon() + "] " + this.description + "(by: " 
        + this.deadline + ")";
```
Perhaps splitting the code into a newline would be useful 😃

jieqiboh reviewed (commented):
LGTM 😄 
Although there were a few minor coding standard violations, I found your code easy to read for the most part, and the current issues should be fairly easy to fix.

Commit:
Implement automatic saving and loading of task list to/from hard disk

youdonnnn committed:
Merge pull request #1 from youdonnnn/branch-Level-7

Implement automatic saving and loading of task list to/from hard disk

Commit:
Implement date and time system for deadlines and events

youdonnnn committed:
Merge pull request #2 from youdonnnn/branch-Level-8

Implement date and time system for deadlines and events

Commit:
Move related functionalities into separate classes for better organisation and readability.

youdonnnn committed:
Merge pull request #3 from youdonnnn/branch-A-MoreOOP

Move related functionalities into separate classes for better organis…

Commit:
Create package to hold all Java classes

chiralcentre commented on src/main/java/youdon/Deadline.java:
```diff
@@ -0,0 +1,24 @@
+        String formattedDateTime = this.deadline.format(formatter);
+        return "[" + this.getTypeIcon() + "][" + this.getStatusIcon() + "] " + this.description + "(by: " + formattedDateTime + ")";
+    }
+}
```
Should String.format be used instead of concatenation to make the code cleaner? I noticed this nit for the other classes as well.

chiralcentre commented on src/main/java/youdon/Parser.java:
Should a switch statement be used to clearly highlight all the different logic branches?

chiralcentre commented on src/main/java/youdon/Storage.java:
Should the return type of loadData() be TaskList instead of ArrayList<Task>?

chiralcentre commented on src/main/java/youdon/Youdon.java:
Should the business logic in parser.parse() be in the main class here instead? I was thinking it doesn't make much sense for the parser to handle the business logic for the execution of the application; instead the parser should only be responsible for extracting the command for each line of input.

chiralcentre commented on src/main/java/youdon/validCommands.java:
Would it be good to abstract out the command types into separate classes as per the requirements of the A-MoreOOP increment?

chiralcentre reviewed (commented):
LGTM, but I have highlighted some issues, which I hope will be of use to you.

Commit:
Merge branch 'add-gradle-support'

Commit:
Add Gradle to automate project builds

Commit:
Add JUnit tests to check each task class

Commit:
Add JavaDoc comments to all public methods and classes

Commit:
Tweak the code to comply with a coding standard

Commit:
Add "find" command for users to find a task by searching for a keyword

youdonnnn committed:
Merge pull request #4 from youdonnnn/branch-Level-9

Add "find" command for users to find a task by searching for a keyword

Commit:
Resolve merge conflicts from merging branch-A-CodingStandard into master

Commit:
Resolve merge conflicts from merging branch-A-JavaDoc into master

Commit:
Rectify some Collection names that were not plural

Commit:
Fix extra import statements that were not removed during merge conflict resolution

Commit:
Fix coding style violations using CheckStyle

youdonnnn committed:
Merge pull request #6 from youdonnnn/branch-A-CheckStyle

Fix coding style violations using CheckStyle

Commit:
Implement basic GUI functionality

Commit:
Resolve some warnings in Ui.java

Commit:
Fix bugs in GUI implementation

youdonnnn committed:
Merge pull request #8 from youdonnnn/branch-Level-10

Branch level 10

Commit:
Add assertions to check basic assumptions

Commit:
Refactor code to improve code quality

There are many chunks of duplicated code after creation of GUI.

Removing duplicated code de-clutters the program and increases readability.

Commit:
Refactor code to improve code quality

There were many redundancies in Ui.java consisting of unused and clunky code. Removal of the unused code and refactoring the clunky code allows for better readability and easier understanding.

youdonnnn committed:
Merge pull request #9 from youdonnnn/branch-A-Assertions

Add assertions to check basic assumptions

youdonnnn committed:
Merge pull request #10 from youdonnnn/branch-A-CodeQuality

Branch a code quality

Commit:
Add snooze feature to snooze deadlines by 1 day each time

The snooze feature allows users to snooze their deadline tasks, allowing more flexbility in task management.

youdonnnn committed:
Merge pull request #11 from youdonnnn/branch-BCD-Extension

Add snooze feature to snooze deadlines by 1 day each time

Commit:
Add name of YoudonBot to the titlebar of the GUI

Commit:
Finalise features and documentation of YoudonBot

Commit:
Resolve most warnings in YoudonBot's program and push all relevant dependencies

Commit:
Add a User Guide for YoudonBot

Commit:
Fix formatting issue in User Guide

Commit:
Remove Duke readme file

Commit:
Fix javadoc comments

Commit:
Fix javadoc comments

Commit:
Fix javadoc comments

Commit:
Fix javadoc comments

Commit:
Fix javadoc comments
